src/Nox.Reference.MacAddresses/AppMacAddressExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Nox.Reference.Common;
+using Nox.Reference.MacAddresses.Models;
+
+namespace Nox.Reference.MacAddresses;
+
+public static class AppMacAddressExtensions
+{
+    private const string ResourceName = "Nox.Reference.MacAddresses.json";
+
+    public static IServiceCollection AddMacAddresses(this IServiceCollection services)
+    {
+        services.AddSingleton(_ => CreateMacAddressService());
```

Comment:
why singleton?

Reply:
The idea of such services is to store data in memory. So, if we created scope or transient instance it would read data from file for each invocation.

Reply:
But you can just read this data one time in static memory.

Reply:
What do you propose is use static class instead of  service? Not sure that's good idea. It will get the same singleton however it will add complexity to handle it in case any dependency appears.

Reply:
Agree with Andey on this one. Although I think we should change all services from loading all data from json to a singe normalized Sqlite db and use Dapper or EF to load and memcache queries for the dll API's. We would however still have to track connection data as a singleton ayway

Reply:
@AndreyDegtyarev No, I didn't propose to use a static class, only having a static field inside a normal class, which population will be handled as a singleton and the class itself could be non-singleton. This doesn't really matter too much right now because the implementation of this service is very simple and we won't have any concurrency or other types of issues though.